rag/rag-base/src/commonMain/kotlin/ai/koog/rag/base/files/FileSystemProvider.kt

```diff
@@ -4,240 +4,241 @@ import kotlinx.io.Sink
 
     /**
-     * Interface defining methods for serializing and deserializing file paths and handling related metadata.
-     * This is a generic interface that operates with paths of a specified type.
+     * Handles serialization and deserialization of file paths.
      */
+    @Deprecated("For internal use only.")
     public interface Serialization<Path> {
+
         /**
-         * Converts the given path to a string representation of the file path.
-         *
-         * @param path The path to convert into a string.
-         * @return A string representation of the supplied path.
-         * @deprecated Use toAbsolutePathString instead for generating string representations of absolute paths.
-         */
-        @Deprecated("Use toAbsolutePathString instead", ReplaceWith("toAbsolutePathString(path)"))
-        public fun toPathString(path: Path): String
-        /**
-         * Converts a given Path object into its absolute path representation as a string.
+         * Converts a [path] to its absolute path string representation.
          *
-         * @param path The Path object to be converted into an absolute path string.
-         * @return The absolute path string representation of the provided Path object.
+         * @param path The path to convert.
+         * @return Absolute path as a string.
          */
         public fun toAbsolutePathString(path: Path): String
+
         /**
-         * Converts a string representation of an absolute file path into a Path object.
+         * Creates a [Path] object from an absolute path string.
          *
-         * @param path The absolute path as a string.
-         * @return A Path object representing the absolute path.
+         * @param path The absolute path string to convert.
+         * @return A path object representing the absolute path.
+         * @throws IllegalArgumentException if the resolved path is not absolute.
          */
         public fun fromAbsoluteString(path: String): Path
+
         /**
-         * Converts a relative path string to a `Path` object by resolving it against a base path.
+         * Resolves a [path] string against a [base] path.
          *
-         * @param base The base `Path` against which the relative path will be resolved.
-         * @param path The relative path in string form to be resolved.
-         * @return A `Path` object representing the resolved path.
+         * @param base The base path for resolution.
+         * @param path The path string to resolve.
+         * @return The resolved path object.
+         * @throws IllegalArgumentException if [path] is an absolute path.
          */
         public fun fromRelativeString(base: Path, path: String): Path
 
         /**
-         * Retrieves the name of the file or directory represented by the given path.
+         * Gets the name component of a [path].
+         * This method works with the path structure and doesn't check if the path actually exists in the filesystem.
```

Comment:
minor: I like the comment about presence on the FS, I'd add it to toAbsolutePathString/fromAbsoluteString/fromRelativeString if it is true there